### ISSUE TITLE
MSAL Angular query string is lost after redirected back. 

### ISSUE THREAD

**Issue by fhljys** [AzureAD/microsoft-authentication-library-for-js#3242]
## Library

- [x] `msal@1.4.0` or `@azure/msal@1.x.x`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-node@1.x.x`
- [ ] `@azure/msal-react@1.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angular@2.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Framework

- [x] Angular 8
- [ ] React
- [ ] Other

## Description

In Angular APP, hit exactly the same issue as described here. 
The query string appeared for a while and then got lost. 
https://stackoverflow.com/questions/63707766/capturing-query-string-before-angular-constructor-is-called

We are using 1.4.0, tried 1.4.8, same issue

## MSAL Configuration
  {
        auth: {
          clientId: clientId,
          authority: AAD_ENDPOINT + (this._tenantId || 'organizations'),
          validateAuthority: true,
          redirectUri: window.location.origin,
          postLogoutRedirectUri: window.location.origin + baseHref,
          navigateToLoginRequestUrl: true
        },
        cache: {
          cacheLocation: window.navigator.userAgent.indexOf('Edge') > -1 ? 'localStorage' : 'sessionStorage'
        },
        system: {
          loadFrameTimeout: 30 * 1000
        },
        framework: {
          isAngular: true
        }
      }
```js
// Provide configuration values here.
// For Azure B2C issues, please include your policies.
```

## Reproduction steps

```js
// Provide relevant code snippets here.
// For Azure B2C issues, please include your policies.
```

## Identity Provider

- [x] Azure AD
- [ ] Azure B2C Basic Policy
- [ ] Azure B2C Custom Policy
- [ ] ADFS
- [ ] Other

## Browsers/Environment

- [x] Chrome
- [x] Firefox
- [x] Edge
- [ ] Safari
- [ ] IE
- [ ] Other (Please add browser name here)

## Regression

- [ ] Did this behavior work before?
Version:

## Security

- [ ] Is this issue security related?

## Source

- [x] Internal (Microsoft)
- [ ] Customer request

**Comment by tnorling** [comment]
@fhljys Can you please share steps to reproduce and verbose level logs from msal? It's unclear from the stackoverflow issue linked, is this an issue with your own custom query string or the hash that contains the tokens returned from AAD? Can you clarify?

**Comment by fhljys** [comment]
Hi @tnorling , thanks for your prompt reply. 

Here is to provide more details
1. I start a page with https://atlas-preview.cloudapp.net/en-us/authoring?**factory=xxx**, 
2. I was been redirected to login.microsoft and when redirect back I got https://atlas-preview.cloudapp.net#id_token=<mytoken>, note, all the path and query string are lost
3. Then at some point I got   https://atlas-preview.cloudapp.net/en-us/authoring?factory=xxx#id_token=<mytoken>. This looks perfect to me. All the query and path are recovered
4. Then I got https://atlas-preview.cloudapp.net, noting is left..(But at this moment, my app is already logged in.

I shared you a gif in teams as well.

**Comment by fhljys** [comment]
@tnorling , after more investigation, I think we found the root cause. 
Issue happens here: 
![image](https://user-images.githubusercontent.com/18674958/112816545-1a9de300-90b4-11eb-9254-b4c3c54711cc.png)
The original code removed hash, but an extra # is left. Angular will treat this as a location change, so it will cancel the original navigation and trigger a new one. The cancel operation somehow triggered an extra pop state. 

So the issue here is why MSAL provides me an extra # after the navigation? Could you fix that?

**Comment by jasonnutter** [comment]
@fhljys Please update to the latest version of `msal`, as this issue (extra `#` remaining in url) was addressed in 1.4.2: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/changelog.md#142

**Comment by fhljys** [comment]
Thanks @jasonnutter. Verified. It works.
